musica/micm/micm.F90

```diff
@@ -1,113 +1,200 @@
 module micm
-   ! Wrapper for MICM functionality
+  use iso_c_binding
 
-   use iso_c_binding
+  ! Note: "micm_core" is included in an external pre-built MICM library that the host
+  ! model is responsible for linking to during compilation
+  use micm_core, only: micm_t
+  use ccpp_kinds, only: kind_phys
 
-   ! Note: "micm_core" is included in an external pre-built MICM library that the host
-   ! model is responsible for linking to during compilation
-   use micm_core, only: micm_t
-   use ccpp_kinds, only: kind_phys  ! TODO(jiwon) - temporary solution until the framework
-                                    ! can handle the kind conversions automatically
+  implicit none
 
-   implicit none
- 
-   private
-   public :: micm_init, micm_run, micm_final
+  public :: micm_init, micm_run, micm_final
+  private :: convert_to_mol_per_cubic_meter, convert_to_mass_mixing_ratio
 
-   type(micm_t), allocatable :: micm_obj
+  type(micm_t), allocatable :: micm_obj
 
- contains
- 
-   !> \section arg_table_micm_init Argument Table
-   !! \htmlinclude micm_init.html
-   subroutine micm_init(config_path, iulog, errcode, errmsg)
-      ! Initializes MICM objects by creating solvers from configure files
+contains
 
-      character(len=*), intent(in)     :: config_path
-      integer, intent(in)              :: iulog
-      integer, intent(out)             :: errcode
-      character(len=512), intent(out)  :: errmsg
+  !> \section arg_table_micm_init Argument Table
+  !! \htmlinclude micm_init.html
+  subroutine micm_init(config_path, iulog, errcode, errmsg)
+    character(len=*), intent(in)    :: config_path
+    integer, intent(in)             :: iulog
+    integer, intent(out)            :: errcode
+    character(len=512), intent(out) :: errmsg
 
-      errcode = 0
-      errmsg = ''
+    errcode = 0
+    errmsg = ''
 
-      ! Constructs MICM object
-      allocate(micm_obj)
-      micm_obj = micm_t(config_path)
+    allocate(micm_obj)
+    micm_obj = micm_t(config_path)
 
-      ! Creates solver
-      errcode = micm_obj%create_solver()
+    errcode = micm_obj%create_solver()
+
+    if (errcode /= 0) then
+      errmsg = "[fatal] [micm] Failed to create MICM solver. Parsing configuration failed. &
+                Please look over at MICM log file for further information."
+      write(iulog,*) trim(errmsg)
```

Comment:
I think you can remove the write statements in init and run for the error messages - when you return, SIMA will handle printing the error messages so this will result in a duplicate error message.

Reply:
thanks!